planner/core/logical_plan_builder.go

```diff
@@ -3173,15 +3176,38 @@ func unfoldWildStar(field *ast.SelectField, outputName types.NameSlice, column [
 	return resultList
 }
 
-func (b *PlanBuilder) addAliasName(selectFields []*ast.SelectField, p LogicalPlan) (resultList []*ast.SelectField, err error) {
-	if len(selectFields) != len(p.OutputNames()) {
-		return nil, errors.Errorf("lengths of selectFields and OutputNames are not equal(%d, %d)",
-			len(selectFields), len(p.OutputNames()))
+func (b *PlanBuilder) addAliasName(ctx context.Context, selectFields []*ast.SelectField, p LogicalPlan) (resultList []*ast.SelectField, err error) {
+	projOutNames := make([]*types.FieldName, 0, len(selectFields))
+	for _, field := range selectFields {
+		var expr expression.Expression
+		colNameField, isColumnNameExpr := field.Expr.(*ast.ColumnNameExpr)
+		if isColumnNameExpr {
+			colName := colNameField.Name.Name
+			if field.AsName.L != "" {
+				colName = field.AsName
+			}
+			projOutNames = append(projOutNames, &types.FieldName{
+				TblName:     colNameField.Name.Table,
+				OrigTblName: colNameField.Name.Table,
+				ColName:     colName,
+				OrigColName: colNameField.Name.Name,
+				DBName:      colNameField.Name.Schema,
+			})
+		} else {
+			// create view v as select name_const('col', 100);
```

Comment:
The view should be wrapped in a projection when `buildDataSource`, why do we still need this branch?

Reply:
Or this function is just called when we build the view?

Reply:
Yeah. Only add alias name for selectField when creating view(_b.capFlag&canExpandAST_). This is compatible with MySQL.